### PR TITLE
Issue 1230-1 : Use secrets in Pod Yamls.

### LINKED ISSUE #1230: Future: Misc issues in Kubernetes manifests.
**Issue/Feature Description:**
Identify and fix miscellaneous issues in K8s manifests. 
    -  DONE:  Get K8s Master IP before starting deploy (deploy is along running process, which blocks getting the IP ) 
    -  DONE: Add utility script to scale services. 
    -  TODO:  Document the scale feature.
    -  DONE: Correct kafka service name in utility script k8s-cleanup.sh
    -  TODO: Optimize deployment yamls  - remove shell scripts
    -  DONE: Use secrets 
    - TODO: Use Ingress.

**Why this issue to fixed / feature is needed(give scenarios or use cases):**

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by rajat-soda**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1230 

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
kind TESTED
> /kind NOT-TESTED

**Test Report**:

image

**Special notes for your reviewer**:

### CODE REVIEW

**Review by kumarashit — APPROVED** [review]
LGTM

**Comment by rajat-soda** [comment]
@asifdxtreme , @AmitRoushan Please review

File: K8s/soda-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: soda-multi-cloud
+  name: soda-secret
+type: Opaque
+stringData:
+  OS_USERNAME: opensds
+  OS_PASSWORD: opensds@123
```
Comment (AmitRoushan) [inline K8s/soda-secret.yaml:24]: Line feed required

File: K8s/soda-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: soda-multi-cloud
+  name: soda-secret
```
Comment (AmitRoushan) [inline K8s/soda-secret.yaml:20]: Suggestion: "soda-IAM-secret" may be more appropriate.

**Review by AmitRoushan — CHANGES_REQUESTED** [review]
IMO, MongoDB [INITDB](https://github.com/sodafoundation/multi-cloud/blob/master/K8s/mongo-service-0.yaml#L77) username password should be scerets

File: K8s/soda-secret.yaml (hunk 1 above)
Comment (rajat-soda) [reply K8s/soda-secret.yaml]: Done

File: K8s/soda-secret.yaml (hunk 2 above)
Comment (rajat-soda) [reply K8s/soda-secret.yaml:20]: no functional impact, skip

**Review by AmitRoushan — APPROVED** [review]
Looks ok to me

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Pull request merged**